pkg/distributor/ha_tracker.go

```diff
@@ -476,18 +490,33 @@ func (h *haTracker) updateKVStore(ctx context.Context, userID, cluster, replica
 		var ok bool
 		if desc, ok = in.(*ReplicaDesc); ok && desc.DeletedAt == 0 {
 			// If the entry in KVStore is up-to-date, just stop the loop.
-			if h.withinUpdateTimeout(now, desc.ReceivedAt) ||
-				// If our replica is different, wait until the failover time.
-				desc.Replica != replica && now.Sub(timestamp.Time(desc.ReceivedAt)) < h.cfg.FailoverTimeout {
+			if h.withinUpdateTimeout(now, desc.ReceivedAt) {
+				return nil, false, nil
+			}
+		}
+		// If our replica is different, wait until the failover time
+		var electedAtTime, electedChanges int64
+		if desc != nil {
+			electedAtTime = desc.ElectedAt
+			electedChanges = desc.ElectedChanges
+		}
+		if desc != nil && desc.Replica != replica {
```

Comment:
I think we can remove both `desc != nil` conditions moving the code to the previos `ok` section.

(Plus, the _if our replica is different_ comment should be on top of the `desc.Replica != replica` check)